Add details about PubSub integration (#221)
kyma-project/community: sig-and-wg/wg-knative/knative-eventing-adoption.md

* A default PubSub.   `Lowest Priority`
* A PubSub per application
* A PubSub specified for an Event type during registration. `Highest Priority`

**Requirements:**

* API to expose available PubSub in Kyma.

> * Implementation to fetch the information from Event, Application or default to decide which PubSub to use in the same order.

Reviewer: ```suggestion
* Implementation fetching the information from the Event, Application or default to decide which PubSub to use in the same order.
```
Question: what does `default` mean here? Does it mean default information?

* A default PubSub.   `Lowest Priority`
* A PubSub per application
* A PubSub specified for an Event type during registration. `Highest Priority`

**Requirements:**

* API to expose available PubSub in Kyma.

**Flow**
![](assets/decide-which-pubsub.svg)

> 

Reviewer: Let's add this introductory line above the diagram:

`The diagram presents the process of deciding which PubSub to use.`

## When to create a channel (PubSub resources)

### Extensibility use case

> A serverless is triggered by an event sent from  application.

Reviewer: ```suggestion
The Event sent from the application triggers serverless.
```

* Get a channel when a subscription is created. If channel does not exist, create the channel.

> * Only create the channel when someone actually wants to consume events.

Reviewer: ```suggestion
* Create the channel only when someone actually wants to consume events.
```

> * This can be extended in a way that we discard the events at an early stage when there is no consumer configured.

Reviewer: ```suggestion
* This can be extended to allow discarding of the events at an early stage when there is no consumer configured.
```

* This will be the default behavior.

### Integration with external messaging systems

The events sent from an application are published to a cloud PubSub. This could trigger an action (compute) in the cloud ecosystem.

* In such a case, while registering an event type, it can be specified if the backing resources(e.g. Knative Channel) should be created or not.

> * This way the user is making a conscious choice to create PubSub resources while registering events.

Reviewer: ```suggestion
* This way the user can make a conscious choice to create PubSub resources while registering events.
```

**Requirements**

> * Kyma eventing needs to listen for the event registeration.

Reviewer: ```suggestion
* Kyma eventing needs to listen for the event registration.
```

   | A consistent experience for external and internal events.        | Even if we do the tranlation at the gateway for events from external, the internal events would still need to deal with Knative APIs |
   | API ensures that the event paylaod is a valid json               | No validation of event payload being a json. This can lead to difficult to troubleshoot errors especially during consumption.        |

   * Eventually, publishing should evolve into a merge proxy that is only mapping some headers or performing an `http` rewrite. The first step in this direction is to evolve the API to align with CE specification. 

  * Consumption

> 	* There is not much difference for consumption apart from the extra knowledge of channels and translating them to Kyma concepts of `event types` and `application identifier`.

Reviewer: ```suggestion
	* There is not much difference for consumption apart from the extra knowledge of channels and translating them to Kyma concepts of `Event types` and `application identifier`.
```

> 	* Consumption needs to implement `Event Activation` as Knative eventing has no such or a similar concept. After the discussion with Knative community, they do not want to introduce such constraints and expect applications to build them.

Reviewer: ```suggestion
	* Consumption needs to implement `Event Activation` as Knative eventing has no such or a similar concept. The Knative community is against introducing such constraints and expect applications to build them.
```
